components/places/src/ffi.rs

```diff
@@ -439,39 +437,40 @@ impl PlacesConnection {
         get_direct_children: bool,
     ) -> Result<Option<BookmarkItem>> {
         self.with_conn(|conn| {
-            let bookmark = bookmarks::public_node::fetch_bookmark(conn, guid, get_direct_children)?;
+            let bookmark = bookmarks::fetch::fetch_bookmark(conn, guid, get_direct_children)?;
             Ok(bookmark.map(BookmarkItem::from))
         })
     }
 
     fn bookmarks_get_all_with_url(&self, url: Url) -> Result<Vec<BookmarkItem>> {
         self.with_conn(|conn| {
-            let bookmarks = bookmarks::public_node::fetch_bookmarks_by_url(conn, &url)?
+            // XXX - We should move the exact type up to the consumers!
```

Comment:
I think these changes are better done in a followup though as they will be breaking (but an improvement!) for consumers (ie, currently they would need to explicitly ignore or otherwise handle folders and separators being returned here even though they never will be)